lib/bloc/account/account_bloc.dart

```diff
@@ -200,19 +199,67 @@ class AccountBloc {
           });
         });    
     }
-  
+
+    void _listenFilterChanges(BreezBridge breezLib) {
+      _paymentFilterController.stream.listen((filter) {
+        _refreshPayments(breezLib);
+      });
+    }
+
     void _refreshPayments(BreezBridge breezLib) {
+      DateTime _firstDate = DateTime(2018);
       if (MockPaymentInfo.isMockData) {
-        _paymentsController.add(MockPaymentInfo.createMockData());
+        List<PaymentInfo> _paymentsList = _filterMockDataPayments(MockPaymentInfo.createMockData());
+        if(_paymentsList.length > 0){
+          _firstDate = DateTime.fromMillisecondsSinceEpoch(_paymentsList.elementAt(0).creationTimestamp.toInt() * 1000);
+        }
+        _paymentsController.add(PaymentsModel(_paymentsList, _paymentFilterController.value, _firstDate));
         return;
       }
 
        breezLib.getPayments().then( (payments) {
-        _paymentsController.add(payments.paymentsList.map( (payment) => new PaymentInfo(payment, _currentUser.currency)).toList());
+         if(payments.paymentsList.length > 0){
+           _firstDate = DateTime.fromMillisecondsSinceEpoch(payments.paymentsList.elementAt(0).creationTimestamp.toInt() * 1000);
+         }
+         _paymentsController.add(PaymentsModel(_filterPayments(payments), _paymentFilterController.value, _firstDate));
       })
-      .catchError(_paymentsController.addError); 
-    }  
-  
+      .catchError(_paymentsController.addError);
+    }
+
+  _filterMockDataPayments(List<PaymentInfo> paymentsList) {
```

Comment:
We don't need this function. use the same function _filterPayments for the real data and mock data.
You will need to change its prototype to get List<PaymentInfo> but it worth the value of removing this function and unifying the flows.